File: travelAgency/Services/src/main/java/services/IServices.java
package services;


import app.model.Flight;
import app.model.User;

import java.sql.SQLException;
import java.time.LocalDate;

public interface IServices {

    void findUserByCredentials(String username, String password, IObserver userObs) throws MyException;

    boolean signUp(String username, String password);

    Iterable<Flight> getAllFlights() throws MyException;

    Iterable<Flight> getFlightsByDestinationDate(String dest, LocalDate date) throws MyException;

    void buyTicket(Flight flight, String client, String address, String tourists, String noSeats) throws MyException, SQLException;

    void logOut(User user, IObserver userObs) throws MyException;
}

File: Anul_2/Semestrul_2/Medii de proiectare si programare/mpp-proiect-csharp-andrea-muresan/TravelAgencyProject/Services/IServices.cs
using System;
using System.Collections.Generic;
using Model;

namespace Services
{
    public interface IServices
    {
        void LogInVerify(string username, string password, IObserver client);

        IEnumerable<Flight> GetAllFlights();

        IEnumerable<Flight> GetFlightsByDateDestination(string destination, string inputdate);

        void buyTicket(int flight_id, string name, string address, string tourists, string noSeats);

        Flight? FindFlight(int id);

        string SignUp(string username, string password);
        void logout(User currentUser, IObserver clientCtrl);
    }
}